src/Compilation.zig

```diff
@@ -2990,6 +3004,75 @@ pub fn generateBuiltinZigSource(comp: *Compilation, allocator: *Allocator) ![]u8
     return buffer.toOwnedSlice();
 }
 
+fn addBuildpkgs(gpa: *Allocator, arena: *Allocator, dir: Directory, name: []const u8, pkg: *Package) anyerror!void {
+    var pkg_dir = Directory {
+        .path = try dir.join(gpa, &[_][]const u8 { name }),
+        .handle = try dir.handle.makeOpenPath(name, .{}),
+    };
+    defer pkg_dir.handle.close();
+    var sub_pkg_it = pkg.table.iterator();
+    while (sub_pkg_it.next()) |sub_pkg| {
+        try addBuildpkgs(gpa, arena, pkg_dir, sub_pkg.key, sub_pkg.value);
+    }
+
+    const source = try generateBuildpkgsZigSource(gpa, pkg);
+    defer gpa.free(source);
+    try pkg_dir.handle.writeFile("buildpkgs.zig", source);
+
+    try pkg.table.put(arena, "buildpkgs", Package.create(gpa, pkg_dir.path, "buildpkgs.zig") catch |err| {
+        fatal("Failed to create buildpkgs Package at path {s}: {s}", .{ pkg_dir.path, @errorName(err) });
+    });
+}
+
+pub fn generateBuildpkgsZigSource(allocator: *Allocator, pkg: *const Package) ![]u8 {
+    const tracy = trace(@src());
+    defer tracy.end();
+
+    var buffer = std.ArrayList(u8).init(allocator);
+    defer buffer.deinit();
+
+    try buffer.writer().writeAll(
+        \\
+        \\pub const names = &[_][]const u8 {
+        \\
+    );
+    //std.log.info("writePackageNames: begin (path={s}, dir={s})", .{
+    //    pkg.root_src_path,
+    //    if (pkg.root_src_directory.path) |p| p else "<none>",
+    //});
+    var pkg_it = pkg.table.iterator();
+    while (pkg_it.next()) |sub_pkg| {
+        //std.log.info("writePackageNames: name={s}", .{sub_pkg.key});
+        try buffer.writer().print("    \"{s}\",\n", .{sub_pkg.key});
+    }
+    try buffer.writer().writeAll(
+        \\};
+        \\
+        \\// using .Inline to force this to be comptime
```

Comment:
See the doc comment in `test/standalone/buildpkgs/missing-comptime/build.zig`.

> //! Currently the user MUST specify comptime when calling "buildpkgs.has"
//! until https://github.com/ziglang/zig/issues/425 is implemented.
//! This build.zig file tries to call "buildpkgs.has" without comptime and the
//! test is to make sure it produces a compile error.
//!
//! Note that the reason why "buildpkgs.has" must be evaluated at comptime is
//! because it will always surround an @import statement.  The problem is
//! that if they forget to add "comptime" to their call, then their build.zig
//! file will "sometimes work" so long as they are building with the necessary
//! packages configured, but then it will fail once the @import is missing
//! which defeats the whole purpose of providing "buildpkgs.has" in the first place.
//!